### PR TITLE
CRT build updates

### PR DESCRIPTION
**Pull request by DanStough**
High-level summary of changes:
* Added a CODEOWNERS file, per rel-eng suggestion
* Bumped the go version to 1.21. I'm also using the new syntax to specify the exact toolchain in the mod file instead go using `.go-version`.
* Added the GHA for building CRT artifacts.
* Moved the Dockerfile to the project root, which is a CRT requirement. I figured that distroless was sufficient and we didn't need both alpine and debian.
* gutted the makefile to make it simpler for local development. Much of the prior functionality is offloaded to CRT. 

Adding the release HCL will be done in another PR.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -33,7 +33,7 @@ func main() {
 
 	// Asking for the version?
 	if *versionFlag {
-		fmt.Fprintln(stderrW, version.HumanVersion)
+		fmt.Fprintln(stdoutW, version.HumanVersion)
```
Comment (t-eckert) [inline main.go:36]: Good fix

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+ARG TARGETARCH
+ARG PRODUCT_VERSION
+
+LABEL name="http-echo" \
+      maintainer="HashiCorp Consul Team <consul@hashicorp.com>" \
+      vendor="HashiCorp" \
+      version=$PRODUCT_VERSION \
+      release=$PRODUCT_VERSION \
+      summary="A test webserver that echos a response. You know, for kids." 
+
```
Comment (t-eckert) [inline Dockerfile:18]: Should we expose a default port and the "ECHO_TEXT" environment variable?

**Review by t-eckert — APPROVED** [review]
Thank you for doing this!

File: Dockerfile (hunk 2 above)
Comment (DanStough) [reply Dockerfile:18]: Yep, I like it.

Comment (DanStough) [reply Dockerfile:18]: I'm going to add a default of `hello-world`, which shouldn't break anyone passing these args already, but would make it easier for quick and dirty testing.

**Pull request merged**